linked_list.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+
+
+class Node(object):
+    """Construct Node object."""
+
+    def __init__(self, val):
+        """Initialize node object."""
+        self.val = val
+        self.next = None
+
+    def get_val(self):
+        """Get data."""
+        return self.val
+
+
+class LinkedList(object):
+    """Handle creation of a linked list."""
+
+    def __init__(self, itr=None):
+        """Init linked list object with optioanl itterable."""
+        self.head = None
+        self.itr = itr
+        if itr:
+            try:
+                for i in itr:
+                    self.insert(i)
+            except TypeError:
+                print('value is not an interable')
+
+    def insert(self, val):
+        """Insert new node at head of list."""
+        new_node = Node(val)
+        new_node.next = self.head
+        self.head = new_node
+
+    def search(self, val):
+        """Search for value and return node."""
+        current = self.head
+        found = False
+        while current and not found:
+            if current.val == val:
+                found = True
+                return current.val
```

Comment:
You want to return the node (`current`), not the value. From the assignment: search(val) will _**return the node containing ‘val’**_ in the list, if present, else None. Also, do you really need `found` here? `return` will break the while loop.